core/src/test/java/org/infinispan/xsite/irac/IracTombstoneCleanupTest.java

```diff
@@ -0,0 +1,376 @@
+import org.infinispan.remoting.rpc.RpcOptions;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.ResponseCollector;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.fwk.TransportFlags;
+import org.infinispan.util.AbstractDelegatingRpcManager;
+import org.infinispan.xsite.status.TakeOfflineManager;
+import org.testng.annotations.Test;
+
+import net.jcip.annotations.GuardedBy;
+
+/**
+ * Basic tests for IRAC tombstone cleanup
+ *
+ * @since 14.0
+ */
+@Test(groups = "xsite", testName = "xsite.irac.IracTombstoneCleanupTest")
+public class IracTombstoneCleanupTest extends MultipleCacheManagersTest {
+
+   private static final String CACHE_NAME = "xsite-tombstone";
+   private static final String SITE_NAME = "LON-1";
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      TransportFlags flags = new TransportFlags().withPortRange(0).withSiteName(SITE_NAME).withFD(true);
+      createClusteredCaches(3, CACHE_NAME, cacheConfiguration(), flags);
+      for (Cache<?, ?> cache : caches(CACHE_NAME)) {
+         // stop automatic cleanup to avoid adding random events to the tests
+         tombstoneManager(cache).stopCleanupTask();
+         extractComponent(cache, TakeOfflineManager.class).takeSiteOffline("NYC");
+      }
+   }
+
+   private static ConfigurationBuilder cacheConfiguration() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.clustering().cacheMode(CacheMode.DIST_SYNC);
+      builder.clustering().hash().numOwners(2).numSegments(16);
+      builder.sites().addBackup().site("NYC").strategy(BackupConfiguration.BackupStrategy.ASYNC);
+      return builder;
+   }
+
+   public void testPrimaryOwnerRoundCleanupsBackup(Method method) {
+      final String key = k(method);
+      final int segment = getSegment(key);
+      Cache<String, String> pCache = findPrimaryOwner(segment);
+      Cache<String, String> bCache = findBackupOwner(segment);
+
+      IracMetadata metadata = dummyMetadata(1);
+
+      tombstoneManager(pCache).storeTombstone(segment, key, metadata);
+      tombstoneManager(bCache).storeTombstone(segment, key, metadata);
+
+      assertEquals(1, tombstoneManager(pCache).size());
+      assertEquals(1, tombstoneManager(bCache).size());
+
+      RecordingRpcManager pRpcManager = rpcManager(pCache);
+
+      pRpcManager.startRecording();
+
+      tombstoneManager(pCache).runCleanupAndWait();
+
+      eventuallyEquals(0, () -> tombstoneManager(pCache).size());
+      eventuallyEquals(0, () -> tombstoneManager(bCache).size());
+
+      IracTombstoneCleanupCommand cmd = pRpcManager.findSingleCommand(IracTombstoneCleanupCommand.class);
+
+      assertNotNull(cmd);
+      assertEquals(segment, cmd.getTombstone().getSegment());
+      assertEquals(key, cmd.getTombstone().getKey());
+      assertEquals(metadata, cmd.getTombstone().getMetadata());
+
+      pRpcManager.stopRecording();
```

Comment:
I suggest doing this in an `@AfterMethod(alwaysRun=true)` method instead, so you don't have to worry about a failure in one test making the next test fail as well.